src/net_processing.h

```diff
@@ -9,6 +9,8 @@
 #include "net.h"
 #include "validationinterface.h"
 
+/** Maximum number of outstanding CMPCTBLOCK requests for the same block. */
+static const int MAX_CMPCTBLOCKS_INFLIGHT_PER_BLOCK = 2;
```

Comment:
Likely not, asking all your peers for a copy of the block poses the same network-DoS risks as connecting to hundreds of nodes, which people like to do because they believe it will help (though it usually actually hurts) them get their blocks out faster.

More importantly, if you're a miner and have good peers, I think it'd be somewhat rare for you to receive a third compact block announce before the first can respond to your blocktxn request, at least it will be once we get proper multi-threaded ProcessMessages implemented to respond to blocktxn requests for the latest block in the background (see #10652 for the beginnings of the steps to do so).